### PR TITLE
deletion of badge if calculation invalidated

### PR DESCRIPTION
**Pull request by anas-drissi**

### CODE REVIEW

**Review by bendaamerahm — CHANGES_REQUESTED** [review]

File: src/components/app-top-bar.js [hunk 1]
```diff
@@ -298,9 +322,33 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
...
+                            );
+                            console.log(
+                                'SECURITY Flow Infos Analysis 2:',
+                                securityAnalysisStatusNode?.preContingencyResult
+                            );
                             if (
                                 tabName === StudyView.RESULTS &&
-                                (loadflowNotif || saNotif)
+                                isNodeBuilt(currentNode) &&
+                                ((loadflowNotif &&
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/components/app-top-bar.js]: it's fixed

File: src/components/app-top-bar.js [hunk 2]
```diff
@@ -298,9 +322,33 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
...
                             let label;
+                            console.log(
+                                'Load Flow Infos Result :',
+                                loadFlowInfosNode?.loadFlowResult
+                            );
+                            console.log(
+                                'SECURITY Flow Infos Analysis 1:',
+                                securityAnalysisStatusNode?.postContingencyResults
+                            );
+                            console.log(
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/components/app-top-bar.js]: it's ok

File: src/components/app-top-bar.js [hunk 3]
```diff
@@ -298,9 +322,33 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
                     >
                         {STUDY_VIEWS.map((tabName) => {
                             let label;
+                            console.log(
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/components/app-top-bar.js]: it's fixed

File: src/components/app-top-bar.js [hunk 4]
```diff
@@ -298,9 +322,33 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
...
+                                'Load Flow Infos Result :',
+                                loadFlowInfosNode?.loadFlowResult
+                            );
+                            console.log(
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/components/app-top-bar.js]: it's fixed

**Comment by etiennehomer** [comment]
The badge for security analysis results is not deleted when the node is invalidated. 

Other problem: when we run a loadflow, then we invalidate the node (the badge is deleted). And then we run a security analysis, we have '2' in the badge instead of '1'

**Review by FranckLecuyer — APPROVED** [review]
Tests: OK
Code review: OK

**Review by bendaamerahm — APPROVED** [review]
code review : ok 
test: ok

**Pull request merged**